android/src/main/java/com/tanguyantoine/react/MusicControlModule.java

```diff
@@ -263,39 +254,59 @@ synchronized public void setNowPlaying(ReadableMap metadata) {
+                || getCurrentActivity() != null && !getCurrentActivity().isFinishing()) {
             String artwork = null;
-            boolean localArtwork = false;
 
             if(metadata.getType("artwork") == ReadableType.Map) {
                 artwork = metadata.getMap("artwork").getString("uri");
-                localArtwork = true;
             } else {
                 artwork = metadata.getString("artwork");
             }
 
             final String artworkUrl = artwork;
-            final boolean artworkLocal = localArtwork;
-
-            artworkThread = new Thread(new Runnable() {
+            Log.d(TAG,"artworkUrl = " + artworkUrl);
+            new AsyncTask<Void, Void, Void>() {
                 @Override
-                public void run() {
-                    Bitmap bitmap = loadArtwork(artworkUrl, artworkLocal);
-
-                    if(md != null) {
-                        md.putBitmap(MediaMetadataCompat.METADATA_KEY_ART, bitmap);
-                        session.setMetadata(md.build());
+                protected Void doInBackground(Void... params) {
+                    // Looper.prepare();
+                    if (Looper.myLooper() == null)
+                    {
+                        Looper.prepare();
                     }
-                    if(nb != null) {
-                        nb.setLargeIcon(bitmap);
-                        notification.show(nb, isPlaying);
+                    try {
+                        theBitmap = Glide.with(getCurrentActivity())
+                                .asBitmap()
+                                .load(artworkUrl)
+                                .submit(1024,597)
+                                .get();
+                    } catch (final ExecutionException e) {
+                        Log.e(TAG, e.getMessage());
+                    } catch (final InterruptedException e) {
+                        Log.e(TAG, e.getMessage());
                     }
-
-                    artworkThread = null;
+                    return null;
                 }
-            });
-            artworkThread.start();
+                @Override
+                protected void onPostExecute(Void dummy) {
+                    if (null != theBitmap) {
+                        // The full bitmap should be available here
+                        // image.setImageBitmap(theBitmap);
+                        Log.d(TAG, "Image loaded");
+
+                        if (md != null) {
+                            md.putBitmap(MediaMetadataCompat.METADATA_KEY_ART, theBitmap);
+                            // session.setMetadata(md.build());
+                        }
+                        if (nb != null) {
+                            nb.setLargeIcon(theBitmap);
+                            notification.show(nb, isPlaying);
+                        }
+                    };
+                }
+            }.execute();
```

Comment:
You're not cancelling the task when another artwork is required, if I spam `setNowPlaying` with different artworks, I might get a random one at the end. It might be a big problem for short tracks (such as ads, jingles or even sound effects)

Reply:
right.
could you kindly please leave your suggestion here?